### ISSUE TITLE
[PROPOSAL] Buildings

### ISSUE THREAD

**Issue by Muddshoe** [sunflower-land/sunflower-land#441]
Before moving forward, this proposal works under the following assumptions:

1. Progression in the game is either:

🔥 Halvening-prone: Becomes more difficult to create/do over time

💧Halvening-resistant: Stays at same difficulty to create/do over time

2. The assets in the game are currently:

🌼Crops: Halvening-resistant, infinite

🔱NFTs: Halvening-prone, finite

🐓Animals: Halvening-prone, infinite

3. The forms of progression in the game are currently:

🏡More Plots: Halvening-resistant

💰Buy NFTs or Animals: Halvening-prone

⛏Skill Progression: 50/50 (Farming is halvening-resistant, gathering is halvening-prone)

4. People play crypto-games for:

🏃‍♀️Fun: Progression systems, grinding, 

🤑 Money: NFT scarcity, liquidity pools, 

While there is an appeal to the game replicating that type of web3 / crypto-game trend of having an economy completely ran by the players, I worry that this sort of setup could end up making a game that is too dependent on the market, on a select number of whales, or on aggressive investments.

This proposal is moving forward with the presumption that the community and the developers want to make a game that can break out of that mold and become widely accessible to a mainstream audience.  I believe this can be done by introducing systems that are more familiar to them and don’t require too much “on-your-feet” knowledge (having to time the market, halvenings, etc)

However, this proposal also doesn’t plan on completely disregarding the foundations of the game, but instead tries to create a complimentary system that instead of rewarding money, directly rewards time. This is to be done using systems that can be found in basically every other mobile game out there.

Introducing: Buildings. 

Buildings are a new type of asset that are: 1. untradeable (meaning they must be created through progression), 2. halvening-resistant (the requirements for building crafting will adjust with the halvenings), 3. infinite (all farms can make buildings if they meet the crafting requirements), and 4. time-rewarding (they require a lot of resources, though these resource requirements do lower over time). They are meant to add a direct form of time-bound progression similar to increasing your skill levels, and increasing your plot numbers. 

In a nutshell, they’re basically like increasing your plot sizes. Although they work differently, of course.

**This means that players, whenever they might start playing the game, will be able to enjoy these halvening-resistant progression systems**

How do buildings work? 

They’re basically like any other NFT item, aside from their building requirements. Like buildings already in the game (like the barn), some of the buildings that will be proposed here allow players to shop/craft unique items.

Additionally, buildings introduce “time-forced” mechanics. Some buildings can be upgraded, like some NFTs. However, doing so takes a specific amount of time like 12 hours, then 1 day, then 3 days, that kind of stuff. Things that games like Clash of Clans already do. Similarly, crafting some of their items require a set amount of time as well.

**PROPOSED BUILDINGS:**

👺 Tavern: “Meet new friends”
BUILDING REQ: 20 Wood, 25 Stones, 5 SFL, 5 Hammers

EFFECTS: 
Contract 1 Goblin (20 Cauliflowers, 5 SFL, 1 Beer) [30 minutes]
Ferment 1 Beer (5 Wheat) [1 hour]

EXPLANATION: The in-game explanation for this is that you’re basically hiring goblins. These workers will be important for building “manned” buildings. You could introduce a level-up system to this building and have new goblin species’ (or even other creatures like centaurs or whathaveyou!)

⚓️ Merchant Docks: “See what’s in store!”
BUILDING REQ: 50 Wood, 10 SFL, 3 Hammers, 2 Goblins
LEVEL 2: 75 Wood, 5 Hammers [12 hours]
LEVEL 3:  100 Wood, 6 Hammers [1 day]

EFFECTS:
Craft <_____> [inventory is randomized every month OR rotates around a seasonal inventory]. Each level-up unlocks one more shopping slot.

EXPLANATION: The Goblins are there to man some of the stalls or to help out whoever is selling there.

🥚 Incubator: “Science!”

BUILDING REQ: 10 Hammers, 30 Stone, 1 Goblin

EFFECTS:
Craft Egg Goop (100 Eggs) [2 hours]
Create Chicken (1 Egg Goop) [1 day]

EXPLANATION: Let the Incubator be a hand-cranked building or something LOL. Though if not then you can just remove the Goblin.

⛏ Quarry

BUILDING REQ: 50 Stone, 25 SFL, 1 Goblins, 5 Hammers[adds 1 mineable stone]
LEVEL 2: 1 Goblin, 100 Stones, 10 Hammers [adds 1 more mineable stones] [12 hours]
LEVEL 3: 150 Stones [adds 1 more mineable stones] [1 day] 
LEVEL 4: 250 Stones, 5 Goblins, 5 Iron [adds 1 mineable iron ore] [2 days]

EFFECTS: 
Each level of the quarry introduces a new mineable resource for the players to destroy. The Goblins are there to mine the quarry and find these ores

These are just some of the ideas, and they’re pretty scattered just to show the potential of buildings as a whole. Of course, everything is subject to change. What do you guys think?

TL;DR: Buildings are a progression system like increasing plot sizes; like so, they are halvening-resistant (crafting requirements lower per halvening) and can be enjoyed by players who might start playing years and years from now when multiple halvenings have already happened. The goal of buildings is to ensure that the game can gather a level of mainstream legitimacy by being playable years after its release.

EDIT 1: For people who are interested in the money-aspect of the game, I believe that by making the game more accessible to a wider audience your assets would inadvertently become more valuable.

**Comment by adamhannigan** [comment]
This is similar to the direction we want to take. 

The plan is to introduce features like this, so that when the halvening occur players can still collect SFL and required resources I agree with a lot of these ideas, the issue is mainly just the time it takes to build out these sort of concepts. It is something that we will need to commit to post-launch alongside the land expansions. The time frame on something as you mentioned could be 6+ months

**Comment by adamhannigan** [comment]
Closing for inactivity. There will be some big redesigns coming in land expansion where we can continue this conversation ❤️
